.ci/openshift-ci/cluster/install_kata.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script installs the built kata-containers in the test cluster,
+# and configure a runtime.
+
+scripts_dir=$(dirname $0)
+deployments_dir=${scripts_dir}/deployments
+configs_dir=${scripts_dir}/configs
+
+source ${scripts_dir}/../../lib.sh
+
+oc project default
+
+worker_nodes=$(oc get nodes |  awk '{if ($3 == "worker") { print $1 } }')
+num_nodes=$(echo $worker_nodes | wc -w)
+[ $num_nodes -ne 0 ] || die "No worker nodes detected. Something is wrong with the cluster"
+
+info "Set installing state annotation and label on all worker nodes"
+for node in $worker_nodes; do
+    info "Deploy kata in worker node $node"
+    oc annotate --overwrite nodes $node \
+	    kata-install-daemon.v1.openshift.com/state=installing
+    oc label --overwrite node $node kata-deploy=true
+done
+
+info "Applying the kata installer daemonset"
+if [ -z "$KATA_INSTALLER_IMG" ]; then
+	# The yaml file uses $IMAGE_FORMAT which gives the
+	# registry/image_stream:image format
+	export KATA_INSTALLER_IMG=$(echo $IMAGE_FORMAT"kata-installer" \
+		| envsubst)
+fi
+envsubst < ${deployments_dir}/daemonset_kata-installer.yaml.in | oc apply -f -
+
+oc get pods
+oc get ds
+ds_pods=($(oc get pods | awk '{if ($1 ~ "kata-deploy") { print $1 } }'))
+cnt=5
+while [[ ${#ds_pods[@]} -gt 0 && $cnt -ne 0 ]]; do
+        sleep 120
+        for i in ${!ds_pods[@]}; do
+                info "Check daemonset ${ds_pods[i]} is running"
+		rc=$(oc exec ${ds_pods[i]} -- cat /tmp/kata_install_status 2> /dev/null)
+                if [ -n "$rc" ]; then
+			info "Finished with status: $rc"
+			oc describe pods ${ds_pods[i]}
+			oc logs ${ds_pods[i]}
+                        unset ds_pods[i]
+		else
+			info "Running"
+		fi
+        done
+        cnt=$((cnt-1))
+done
+
+if [ $cnt -eq 0 ]; then
+	for p in ${ds_pods[@]}; do
+		info "daemonset $p did not finish"
+		oc describe pods $p
+		oc logs $p
+	done
+	# TODO: exit here?
+fi
+
+# Finally remove the installer daemonset
+info "Deleting the kata installer daemonset"
+oc delete -f ${deployments_dir}/daemonset_kata-installer.yaml.in
+
+# TODO: Check the installation succeed.
+
+# Apply the CRI-O configuration
+info "Configuring kata runtime for CRI-O"
+if [ -z "$KATA_CRIO_CONF_BASE64" ]; then
+	export KATA_CRIO_CONF_BASE64=$(echo $(cat $configs_dir/crio_kata.conf|base64) \
```

Comment:
This is starting to look a bit like lisp :)

Reply:
To be honest, I found it a little bit uncommon (so to speak). But I wanted the config files versioned in plain-text so that any change would be properly visualized with diff. So the only solution I've found was to convert from text to base64 on run-time. Perhaps the `if [ -z "$KATA_CRIO_CONF_BASE64" ]; then` is not needed....